src/NodaTime.Web/Startup.cs

```diff
@@ -129,5 +136,40 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
             // Force the set of releases to be first loaded on startup.
             app.ApplicationServices.GetRequiredService<IReleaseRepository>().GetReleases();
         }
+
+        /// Sets the Cache-Control header for static content, conditionally allowing the browser to use the content
+        /// without revalidation.
+        private void SetCacheControlHeaderForStaticContent(IHostingEnvironment env, HttpContext context)
+        {
+            var headers = new ResponseHeaders(context.Response.Headers);
+
+            if (headers.ContentType.IsSubsetOf(TextHtml))
+            {
+                // Don't set Cache-Control for HTML files (e.g. /tzdb/). The browser can figure out when to revalidate.
+                // (Which it can do easily, since we send an ETag with all static content responses).
+                return;
+            }
+
+            // Otherwise if the request was made with a file version query parameter (?v=..., as sent by
```

Comment:
Given that the logic is only about setting MaxAge as far as I can tell, I'd probably do that just with:

```csharp
var maxAge =
    context.Request.Query.ContainsKey("v") ? TimeSpan.FromDays(365)
    : env.IsDevelopment ? TimeSpan.FromMinutes(2)
    : TimeSpan.FromDays(1);

headers.CacheControl =...;
```

(With an explanatory comment, of course.) But this is fine too.